### ISSUE TITLE
Error: [$injector:unpr] Unknown provider: $stateProvider,$urlRouterProvider

### ISSUE THREAD

**Issue by rustanacexd** [yeoman/generator-angular#902]
I am having that error and I can't troubleshoot why - 

I made sure I added dependencies,

here is my app.js

``` javascript
angular
  .module('myappApp', ['ui.router'])
  .config(['$stateProvider,$urlRouterProvider',
    function($stateProvider, $urlRouterProvider) {
      $urlRouterProvider.otherwise('/');
      $stateProvider
        .state('home', {
          url: '/',
          templateUrl: 'views/main.html',
          controller: 'MainCtrl'
        });
 }
]);
```

I also made sure that angular-ui-router.js is loaded properly. I am stuck :(

**Comment by tamaghna91** [comment]
my problem is not persisted as i have written like this which throws the error:

angular.module('spBlogger.posts').config(['$stateProvider','$locationProvider',
	function($stateProvider,$locationProvider){	
 $stateProvider.state('allPosts',{
 url:'/posts',
 templateUrl: 'modules/posts/views/posts.html',
 controller: 'PostController'
 });
 $stateProvider.state('singlePost',{
 url:'/posts/:id/:permalink',
 templateUrl: 'modules/posts/views/singlePost.html',
 controller: 'PostDetailsController'
 });
}]);


![screenshot from 2015-06-12 14 56 47](https://cloud.githubusercontent.com/assets/8926058/8136815/75a45258-115c-11e5-8d6f-ab91dbd787b6.png)

**Comment by srikanth-chitturi** [comment]
I am having the same error from spblogger project of angularjs Novice to Ninja . any solution for this.

Uncaught Error: [$injector:unpr] Unknown provider: stateProvider <- state

**Comment by Amar3006** [comment]
Try this code. It's working fine for me.
<script type="text/javascript">
        var myApp = angular.module('myApp', ['ui.router']);
        myApp.config(['$stateProvider','$urlRouterProvider',
            function ($stateProvider, $urlRouterProvider) { 
                $stateProvider.
                state('Page1', {
                    url: '/Page1',
                    templateUrl: 'Page1.html'
                }).
                 state('Page2', {
                     url: '/Page2',
                     templateUrl: 'Page2.html'
                 })
            }]);
    </script>

**Comment by christopher-adolphe** [comment]
Hello @srikanth-chitturi and @tamaghna91  I started working the sample project from this ebook and got stuck as well. After reading the code several times I noticed that there was a mistake in the excerpt from the ebook for the code in app.js more precisely in the .run() block. The dollar sign in missing for 'state' which is added as dependency. Rewriting the code as follows solves the issue.

`angular.module('spBlogger').run('$state', function($state) {
        $state.go('allPosts');
});`
